Add helper to compute the recorded luminosity of a sample

blinkseb opened the pull request:
The script automatically runs brilcalc on lxplus with the stored
luminosity sections of a sample, and update the luminosity in the
database for the corresponding sample.

delaere commented on scripts/compute_sample_luminosity.py:
```diff
@@ -0,0 +1,109 @@
+        print('')
+
+        cmds = ['brilcalc', 'lumi', '--normtag', '/afs/cern.ch/user/c/cmsbril/public/normtag_json/OfflineNormtagV1.json', '--output-style', 'csv', '-i', '"%s"' % str(sample.processed_lumi.replace('"', ''))]
+        cmd = 'export PATH="/afs/cern.ch/user/s/sbrochet/.local/bin:/afs/cern.ch/cms/lumi/brilconda-1.0.3/bin:$PATH"; export PYTHONPATH="/afs/cern.ch/user/s/sbrochet/.local/lib/python2.7"; ' + ' '.join(cmds)
```
Is there no way to avoid an explicit use of you home dir?

blinkseb replied:
Unfortunately not, since `brilcalc` must be installed by the user using `pip` before using it ...

swertz replied:
This is just a one-time thing, as a requirement to be done before using this script?

blinkseb replied:
It's a one-time thing, so I did it on lxplus to avoid everyone to do it. But now you mention it, it may not work for everyone. Can you try it?

delaere commented on scripts/compute_sample_luminosity.py:
```diff
@@ -0,0 +1,109 @@
+
+        cmds = ['brilcalc', 'lumi', '--normtag', '/afs/cern.ch/user/c/cmsbril/public/normtag_json/OfflineNormtagV1.json', '--output-style', 'csv', '-i', '"%s"' % str(sample.processed_lumi.replace('"', ''))]
+        cmd = 'export PATH="/afs/cern.ch/user/s/sbrochet/.local/bin:/afs/cern.ch/cms/lumi/brilconda-1.0.3/bin:$PATH"; export PYTHONPATH="/afs/cern.ch/user/s/sbrochet/.local/lib/python2.7"; ' + ' '.join(cmds)
+        ssh_cmds = ['ssh', 'lxplus.cern.ch', cmd]
```
The local and lxplus username must be the same... it would be nice to add an option to specify the remote user. Maybe to be combined with the previous comment.

blinkseb replied:
Good idea

blinkseb commented:
@delaere I've updated the PR to add ability to specify another remote username. But before merging, I'd like to have someone other than me testing the script, I'm afraid it may not work due to the local installation of brilcalc in my home folder...

swertz commented:
I'm getting
```
bash: brilcalc: command not found
Traceback (most recent call last):
  File "./compute_sample_luminosity.py", line 115, in <module>
    main()
  File "./compute_sample_luminosity.py", line 104, in main
    compute_luminosity(sample, options)
  File "./compute_sample_luminosity.py", line 78, in compute_luminosity
    brilcalc_result = subprocess.check_output(ssh_cmds)
  File "/cvmfs/cms.cern.ch/slc6_amd64_gcc491/external/python/2.7.6-cms/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['ssh', 'swertz@lxplus.cern.ch', 'export PATH="/afs/cern.ch/user/s/sbrochet/.local/bin:/afs/cern.ch/cms/lumi/brilconda-1.0.3/bin:$PATH"; export PYTHONPATH="/afs/cern.ch/user/s/sbrochet/.local/lib/python2.7"; brilcalc lumi --normtag /afs/cern.ch/user/c/cmsbril/public/normtag_json/OfflineNormtagV1.json --output-style csv -i "{257531:[[5,45],[50,143]],257735:[[1,15]],257645:[[37,73],[75,1096]],257751:[[1,463]],257599:[[42,118]],256843:[[1,204],[207,284],[286,378],[380,461],[463,587],[598,627],[630,661],[1001,1034],[1036,1081],[1083,1191],[1193,1193],[1195,1329],[1331,1332]],256842:[[131,132]],258158:[[1,1088],[1091,1786]],258157:[[1,56]],256926:[[35,50],[53,62],[64,65]],257614:[[1,16]],258136:[[1,60]],256869:[[1,34]],256868:[[5,33],[35,200],[202,492]],257819:[[1,248]],256866:[[34,47]],256941:[[1,17],[19,29],[103,105],[107,126],[129,129],[131,168],[170,170],[175,290],[293,294]],257682:[[66,366]],257461:[[44,95]],256674:[[1,2]],256675:[[1,106],[111,164]],256676:[[1,160],[162,208]],257968:[[69,326]],257816:[[1,385]],256673:[[55,56]],256867:[[1,16],[19,94]],256801:[[73,263]],256630:[[5,26]],257969:[[1,634]],258129:[[30,124]],257723:[[1,1],[3,108],[114,148]],257722:[[1,19]],257804:[[1,17]],257805:[[1,249]],256677:[[1,291],[293,390],[392,397],[400,455],[457,482]],257613:[[14,1307]]}"']' returned non-zero exit status 127
```

blinkseb commented:
Ok so I've updated the PR. Installing `brilcalc` in my own directory does not seem to work, so I added a new `--bootstrap` option, which is only needed the first time. It installs `brilcalc` on user home folder. Typical usage is now

```bash
# Only the first time
./compute_sample_luminosity.py --bootstrap

# Then
./compute_sample_luminosity.py -i 777
```

@swertz Can you test it please? :)

swertz commented:
It works! Thanks a lot! :)
